### DISCUSSION TITLE
How to query for Plugin Performance metrics

### DISCUSSION THREAD

**Discussion by martywaz** [rabbitmq/rabbitmq-delayed-message-exchange#254] [General]
Hello,

I saw support for this, or something like this, was added for 3.10 version. Can anyone share details of how I can retrieve this information at runtime using C#?

My goal is to be able to query and get how many messages are delayed....and what routing key they will use when fired. For us, the routing key corresponds to specific queues.  With that...i can easily tell my customers that they have N number of messages to retry for Q x.

also, is it possible to remove a specific message or any matching a specific routing key?

The only option previously was to disable and reenable the plugin if you wanted to eliminate a message from retry.  However, this removed everything.

Thanks!

**Comment by michaelklishin** [comment]
I assume you are referring to https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/pull/197.

All metrics related to this exchange are exposed the same way metrics are exposed using any other exchange. They can be collected using [Prometheus](https://rabbitmq.com/prometheus.html) (highly recommended) and various [HTTP API](https://rabbitmq.com/management.html#http-api) clients.

**Reply by martywaz** [reply]
ok...so if I understand correctly, that would give me message rate in and message rate out.  What I'm trying to find is way though to give me the number of messages waiting to be delivered

**Reply by michaelklishin** [reply]
IIRC there is a metric like that, see the exchange page in the management UI and what the request and response are to fetch it (using your browser's Developer Toools).
